homeassistant/components/camera/amcrest.py

```diff
@@ -4,21 +4,156 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/camera.amcrest/
 """
+import asyncio
 import logging
+from requests import RequestException
 
-from homeassistant.components.amcrest import (
-    DATA_AMCREST, STREAM_SOURCE_LIST, TIMEOUT)
-from homeassistant.components.camera import Camera
+import voluptuous as vol
+
+from custom_components.amcrest import (
+    DATA_AMCREST, DATA_AMCREST_LOCK, STREAM_SOURCE_LIST, TIMEOUT)
+from homeassistant.components.camera import (
+    Camera, DOMAIN, SUPPORT_ON_OFF, CAMERA_SERVICE_SCHEMA)
 from homeassistant.components.ffmpeg import DATA_FFMPEG
-from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+from homeassistant.const import (
+    ATTR_ENTITY_ID, CONF_NAME, STATE_ON, STATE_OFF)
+from homeassistant.loader import bind_hass
 from homeassistant.helpers.aiohttp_client import (
     async_get_clientsession, async_aiohttp_proxy_web,
     async_aiohttp_proxy_stream)
+from homeassistant.helpers.service import extract_entity_ids
 
 DEPENDENCIES = ['amcrest', 'ffmpeg']
 
 _LOGGER = logging.getLogger(__name__)
 
+DATA_AMCREST_CAMS = 'amcrest_cams'
+
+OPTIMISTIC = True
+
+_BOOL_TO_STATE = {True: STATE_ON, False: STATE_OFF}
+
+SERVICE_ENABLE_RECORDING = 'amcrest_enable_recording'
+SERVICE_DISABLE_RECORDING = 'amcrest_disable_recording'
+SERVICE_GOTO_PRESET = 'amcrest_goto_preset'
+SERVICE_SET_COLOR_BW = 'amcrest_set_color_bw'
+SERVICE_AUDIO_ON = 'amcrest_audio_on'
+SERVICE_AUDIO_OFF = 'amcrest_audio_off'
+SERVICE_MASK_ON = 'amcrest_mask_on'
+SERVICE_MASK_OFF = 'amcrest_mask_off'
+SERVICE_TOUR_ON = 'amcrest_tour_on'
+SERVICE_TOUR_OFF = 'amcrest_tour_off'
+
+ATTR_PRESET = 'preset'
+ATTR_COLOR_BW = 'color_bw'
+
+CBW_COLOR = 'color'
+CBW_AUTO = 'auto'
+CBW_BW = 'bw'
+CBW = [CBW_COLOR, CBW_AUTO, CBW_BW]
+
+SERVICE_GOTO_PRESET_SCHEMA = CAMERA_SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_PRESET): vol.All(vol.Coerce(int), vol.Range(min=1)),
+})
+SERVICE_SET_COLOR_BW_SCHEMA = CAMERA_SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_COLOR_BW): vol.In(CBW),
+})
+
+_MOT_DET_WINDOW = {False: [{'window': 1, 'sensitive': 75, 'threshold': 12},
+                           {'window': 2, 'sensitive': 50, 'threshold': 16}],
+                   True:  [{'window': 1, 'sensitive': 75, 'threshold':  6},
+                           {'window': 2, 'sensitive': 75, 'threshold':  6}]}
+
+
+@bind_hass
+def enable_recording(hass, entity_id=None):
```

Comment:
I believe all these functions at the top (enable_recording through tour_off) should be removed. I added them because when I started making these changes the file contained functions like these, but I believe there was a system wide change to remove functions like these.